### ISSUE TITLE
Ollinet: Auction bidder did not receive ATOM; bid shows error in logs

### ISSUE THREAD

**Issue by rowgraus** [Agoric/Agoric#7844]
## Describe the bug
Ran happy path scenario in Ollinet
The by-price bid was correctly picked up and handled by the auction, but did not transfer ATOM proceeds to the user

We also see an error associated with bids related to the one at issue when we examine bids from IAVL tree: 
string"Error: key (a string) already registered in collection "addressToUpdater""

## Additional Details
All three bids showed as well formed when listed prior to run.  All bids placed by a test account.
```
agoric-cli git:(dc-oracle-setPrice) bin/agops inter bid list --from test-acct
{"id":"bid-1685075828626","discount":10,"give":{"Bid":"80 IST"},"maxBuy":"1000000 ATOM","result":"Your bid has been accepted"}
{"id":"bid-1685075847435","price":"9 IST/ATOM","give":{"Bid":"90 IST"},"maxBuy":"1000000 ATOM","result":"Your bid has been accepted"}
{"id":"bid-1685075866382","discount":15,"give":{"Bid":"150 IST"},"maxBuy":"1000000 ATOM","result":"Your bid has been accepted"}
```

The expected ATOM filled to each bid should have been:
8.897786676 (Discount 10, IST 80)
10.01001001 (Price 9, IST 90)
16.43290349 (Discount 15, IST 150) (partial fill)

Bids 1 and 3 exited correctly for the user.  Bid 2 did not.

User purse balance for ATOM is 25.330689
It should be ~35.34
Delta is the expected ATOM from the errored bid.  10.01001001

The auction proceeded as if it had sold to all bids accurately.

## To Reproduce
Run Happy Path scenario in ollinet

## Platform Environment
Ollinet-81 
Hash: ea8c1c64911b4c58fb43635b25e17e3d50d0cf2a

**Comment by ivanlei** [comment]
```
2023-05-26T05:01:13.638Z SwingSet: ls: v14: Logging sent error stack (Error#4)
2023-05-26T05:01:13.638Z SwingSet: ls: v14: Error#4: key agoric1ghztytqm90hhlnae3ghxlqfhae3kazjwcfg3z8 already registered in collection addressToUpdater
2023-05-26T05:01:13.638Z SwingSet: ls: v14: Error: key (a string) already registered in collection "addressToUpdater"
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at fail (/bundled-source/.../node_modules/ses/src/error/assert.js:357)
 at doInit (/bundled-source/.../packages/swingset-liveslots/src/collectionManager.js:375)
 at getPurse (.../vats/src/vat-bank.js:483)
 at getPurse (.../vats/src/vat-bank.js:446)
 at apply ()
 at apply ()
 at ()

2023-05-26T05:01:13.639Z SwingSet: ls: v14: Error#4 ERROR_NOTE: Sent as error:liveSlots:v14#70004
2023-05-26T05:01:13.737Z SwingSet: vat: v40: wallet agoric1ghztytqm90hhlnae3ghxlqfhae3kazjwcfg3z8 OFFER ERROR: (RemoteError(error:liveSlots:v14#70004)#12)
2023-05-26T05:01:13.740Z SwingSet: vat: v40: RemoteError(error:liveSlots:v14#70004)#12: key (a string) already registered in collection "addressToUpdater"
2023-05-26T05:01:13.741Z SwingSet: vat: v40: Error: key (a string) already registered in collection "addressToUpdater"
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()
 
```

**Comment by ivanlei** [comment]
```
2023-05-26T05:01:13.744Z SwingSet: vat: v40: wallet agoric1ghztytqm90hhlnae3ghxlqfhae3kazjwcfg3z8 offerStatus { id: 'bid-1685075847435', invitationSpec: { callPipe: [ [ 'makeBidInvitation', [ [Object] ] ] ], instancePath: [ 'auctioneer' ], source: 'agoricContract' }, offerArgs: { maxBuy: { brand: Object [Alleged: ATOM brand] {}, value: 1_000_000_000_000n }, offerPrice: { denominator: { brand: Object [Alleged: ATOM brand] {}, value: 1n }, numerator: { brand: Object [Alleged: IST brand] {}, value: 9n } } }, proposal: { give: { Bid: { brand: Object [Alleged: IST brand] {}, value: 90_000_000n } } }, result: 'Your bid has been accepted', numWantsSatisfied: 1, error: 'Error: key (a string) already registered in collection "addressToUpdater"' }
2023-05-26T05:01:13.756Z SwingSet: xsnap: v40: RemoteError(error:liveSlots:v14#70004)#12 ERROR_NOTE: Sent as error:liveSlots:v40#70022
2023-05-26T05:01:13.756Z SwingSet: ls: v40: Logging sent error stack (RemoteError(error:liveSlots:v14#70004)#12)
2023-05-26T05:01:13.922Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (RemoteError(error:liveSlots:v40#70022)#10)
2023-05-26T05:01:13.924Z SwingSet: xsnap: v10: RemoteError(error:liveSlots:v40#70022)#10: key (a string) already registered in collection "addressToUpdater"
2023-05-26T05:01:13.924Z SwingSet: xsnap: v10: Error: key (a string) already registered in collection "addressToUpdater"
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()

2023-05-26T05:01:14.000Z SwingSet: ls: v9: Logging sent error stack (Error#7)
2023-05-26T05:01:14.001Z SwingSet: ls: v9: Error#7: Object [Alleged: IST payment] {} was not a live payment for brand Object [Alleged: IST brand] {} . It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
2023-05-26T05:01:14.002Z SwingSet: ls: v9: Error: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at fail (/bundled-source/.../node_modules/ses/src/error/assert.js:357)
 at assertLivePayment (.../ertp/src/paymentLedger.js:239)
 at depositInternal (.../ertp/src/paymentLedger.js:265)
 at apply ()
 at apply ()
 at In "deposit" method of (IST Purse purse) (/bundled-source/.../node_modules/@endo/exo/src/exo-tools.js:42)
 at apply ()
 at apply ()
 at dispatchToHandler (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:163)
 at win (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:511)
 at ()

2023-05-26T05:01:14.002Z SwingSet: ls: v9: Error#7 ERROR_NOTE: Sent as error:liveSlots:v9#70007
```

**Comment by ivanlei** [comment]
```
2023-05-26T05:01:14.135Z SwingSet: ls: v14: RemoteError(error:liveSlots:v9#70007)#5 ERROR_NOTE: Sent as error:liveSlots:v14#70005
2023-05-26T05:01:14.256Z SwingSet: vat: v40: wallet agoric1ghztytqm90hhlnae3ghxlqfhae3kazjwcfg3z8 offerStatus { id: 'bid-1685075847435', invitationSpec: { callPipe: [ [ 'makeBidInvitation', [ [Object] ] ] ], instancePath: [ 'auctioneer' ], source: 'agoricContract' }, offerArgs: { maxBuy: { brand: Object [Alleged: ATOM brand] {}, value: 1_000_000_000_000n }, offerPrice: { denominator: { brand: Object [Alleged: ATOM brand] {}, value: 1n }, numerator: { brand: Object [Alleged: IST brand] {}, value: 9n } } }, proposal: { give: { Bid: { brand: Object [Alleged: IST brand] {}, value: 90_000_000n } } }, result: [ { status: 'rejected', reason: [Error: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.] } ], numWantsSatisfied: 1, error: 'Error: key (a string) already registered in collection "addressToUpdater"' }
2023-05-26T05:01:14.269Z SwingSet: ls: v40: Logging sent error stack (RemoteError(error:liveSlots:v14#70005)#13)
2023-05-26T05:01:14.270Z SwingSet: ls: v40: RemoteError(error:liveSlots:v14#70005)#13: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
2023-05-26T05:01:14.271Z SwingSet: ls: v40: Error: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()

2023-05-26T05:01:14.271Z SwingSet: ls: v40: RemoteError(error:liveSlots:v14#70005)#13 ERROR_NOTE: Sent as error:liveSlots:v40#70023
2023-05-26T05:01:14.396Z SwingSet: vat: v7: Temporary logging of sent error (RemoteError(error:liveSlots:v40#70023)#4)
2023-05-26T05:01:14.396Z SwingSet: vat: v7: RemoteError(error:liveSlots:v40#70023)#4: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
2023-05-26T05:01:14.397Z SwingSet: vat: v7: Error: (an object) was not a live payment for brand "[Alleged: IST brand]". It could be a used-up payment, a payment for another brand, or it might not be a payment at all.
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at decodeEntry (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:451)
 at map ()
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:453)
 at map ()
 at decodeEntry (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:451)
 at map ()
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:453)
 at decodeEntry (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:451)
 at map ()
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:453)
 at map ()
 at map ()
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at deliver (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1084)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1509)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()
```

**Comment by dckc** [comment]
A visualization of the race condition.

### Prerequisite: multiple outstanding long-running offers

```mermaid
sequenceDiagram

title Creating a Vault

actor Bidder
participant WalletFactory
participant Zoe
participant Auctioneer

    Bidder->>WalletFactory: bid1
    Bidder->>WalletFactory: bid2
    WalletFactory->>Zoe: bid1
    WalletFactory->>Zoe: bid2
    Zoe->>Auctioneer: bid1
    Zoe->>Auctioneer: bid2
```

### Trigger: multiple offers resolve within a window

Then the timer triggers the auctioneer to award the bidder:

```mermaid
sequenceDiagram

title Creating a Vault

actor Bidder
participant WalletFactory
participant Zoe
participant Auctioneer

    Auctioneer->>Zoe: payout1
    Auctioneer->>Zoe: payout2
    Zoe->>WalletFactory: payout1
    WalletFactory->>VatBank: bank(agoric1).getPurse(ibc/atom)
    VatBank->>VatBridge: ...1...
    Zoe->>WalletFactory: payout2
    WalletFactory->>VatBank: bank(agoric1).getPurse(ibc/atom)
    VatBank->>VatBridge: ...2...
```

When the 2nd `bank(agoric1).getPurse(ibc/atom)` hits `VatBank`, the race condition is hit.

**Comment by ivanlei** [comment]
Reopening for merge into release branch
